CUDADataFormats/Track/interface/TrackSoAHeterogeneousT.h

```diff
@@ -19,39 +19,60 @@ class TrackSoAHeterogeneousT {
   using hindex_type = uint32_t;
   using HitContainer = cms::cuda::OneToManyAssoc<hindex_type, S + 1, 5 * S>;
 
+  // quality accessors
+  constexpr Quality quality(int32_t i) const { return reinterpret_cast<Quality>(quality_(i)); }
+  constexpr Quality &quality(int32_t i) { return reinterpret_cast<Quality &>(quality_(i)); }
+  constexpr Quality const *qualityData() const { return reinterpret_cast<Quality const *>(quality_.data()); }
+  constexpr Quality *qualityData() { return reinterpret_cast<Quality *>(quality_.data()); }
+
+  // chi2 accessors
+  constexpr auto &chi2(int32_t i) { return chi2_(i); }
+  constexpr auto chi2(int32_t i) const { return chi2_(i); }
+
+  // stateAtBS accessors
+  constexpr TrajectoryStateSoAT<S> &stateAtBS() { return stateAtBS_; }
```

Comment:
already discussed. I strongly disagree

Reply:
It is not a question of "special circumstances". Coding rules were written at a time of naive OOD. This is DDD. This is a struct composed of SoAs. A new type of beast. Old rules do not apply.

Reply:
> This was the consequence of applying the coding rule 4.25 which requires only one of each public, protected and private section. (It can be waived under special circumstances, though).

4.25 is `Each class may have only one each of public, protected, and private sections, which must be declared in that order.`

was it something from section 7 instead?

Reply:
Yes, also. Rule 7.4 stipulates that structs should be real PODs with no getters and setters, and that applied here.

Reply:
Never understood why a POD looses its "podness" with trivial getters or setters:
what's the difference btw  constepxr float A::x() const; and constexpr float x(A const&); ? 
Different are constructors or destructor that definitively defy Podness.

Reply:
To me 7.4 (and in particular the linked [C.131 from the C++ Core Guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rh-get)) does not instruct against a struct with public data members and non-setter/getter member functions.

Reply:
> Never understood why a POD looses its "podness" with trivial getters or setters:

With C++ standard definitionsof [POD](https://en.cppreference.com/w/cpp/named_req/PODType) (or now [TrivialType](https://en.cppreference.com/w/cpp/named_req/TrivialType)) "podness" is not lost with regular member functions.

Reply:
Ok. So no reason CMS shall be more restrictive than Standard.